lib/project.js

```diff
@@ -116,6 +126,40 @@ function Project$persist() {
   }
 }
 
+function Project$hasBlip() {
+  return this.optionalDep('sl-blip') !== undefined ||
+    this.script('preinstall') === BLIP_SCRIPT ||
+    this.script('install') === BLIP_SCRIPT ||
+    this.script('postinstall') === BLIP_SCRIPT;
+}
+
+function Project$updateBlip() {
+  // Always write out the file, replacing previous versions if set.
+  var blip = fs.readFileSync(BLIP_SRC_PATH, 'utf8');
+  var scriptsDir = path.resolve(this.rootPath, 'scripts');
+  var dstPath = path.resolve(scriptsDir, 'sl-blip.js');
+  try { fs.mkdirSync(scriptsDir); } catch (e) { }
+  fs.writeFileSync(dstPath, blip);
+
+  // remove from optionalDependencies if present
+  this.optionalDep('sl-blip', false);
+  // it is still detected, but not in optionalDependencies, we're done!
+  if (this.hasBlip()) {
+    return;
+  }
+  // find the first empty *install script and use it
+  if (!this.script('preinstall')) {
+    return this.script('preinstall', BLIP_SCRIPT);
+  }
+  if (!this.script('install')) {
+    return this.script('install', BLIP_SCRIPT);
+  }
+  if (!this.script('postinstall')) {
```

Comment:
I would prefer postinstall to install, if this is order of less-often-used scripts